tests/components/adguard/test_config_flow.py

```diff
@@ -49,13 +49,14 @@ async def test_connection_error(
         exc=aiohttp.ClientError,
     )
 
-    flow = config_flow.AdGuardHomeFlowHandler()
-    flow.hass = hass
-    result = await flow.async_step_user(user_input=FIXTURE_USER_INPUT)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=FIXTURE_USER_INPUT
+    )
 
-    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
-    assert result["errors"] == {"base": "cannot_connect"}
+    assert result
```

Comment:
Very unlikely that result is False instead of None

but shouldn't it be tested explicitly that it's a dict ?

Reply:
Nope. `FlowResult`

Reply:
My point was that the test should check explicitly the expected Object type and not to leave it "unwired" (also applies to other asserts below)

Reply:
That is incorrect according to mypy & pyright. 

It returns a TypeDict called FlowResult, which has those optional. Thus directly accessing is incorrect and wrong according to type checking.

Reply:
```suggestion
    assert isinstance(result, FlowResult)
```

But I'm OK with removing :)